File: frontend/src/lib/exchange.ts
let cachedRate: number | null = null;
let cachedAt = 0;
const CACHE_TTL = 3600_000; // 1 hour

export async function getEurRate(): Promise<number> {
  const now = Date.now();
  if (cachedRate != null && now - cachedAt < CACHE_TTL) {
    return cachedRate;
  }
  try {
    const res = await fetch("https://api.cnb.cz/cnbapi/exrates/daily?lang=EN", {
      next: { revalidate: 3600 },
    });
    const data: { rates: Array<{ currencyCode: string; rate: number; amount: number }> } = await res.json();
    for (const rate of data.rates) {
      if (rate.currencyCode === "EUR") {
        cachedRate = rate.rate / (rate.amount || 1);
        cachedAt = now;
        return cachedRate;
      }
    }
  } catch {
    // fallback
  }
  return 25.0;
}

File: backend/scanner/exchange.py
from __future__ import annotations

import httpx

_eur_rate: float | None = None


def get_eur_rate() -> float:
    global _eur_rate
    if _eur_rate is not None:
        return _eur_rate

    try:
        resp = httpx.get("https://api.cnb.cz/cnbapi/exrates/daily?lang=EN", timeout=10)
        resp.raise_for_status()
        data = resp.json()
        for rate in data.get("rates", []):
            if rate.get("currencyCode") == "EUR":
                _eur_rate = rate["rate"] / rate.get("amount", 1)
                return _eur_rate
    except Exception:
        pass

    _eur_rate = 25.0
    return _eur_rate


def clear_cache() -> None:
    global _eur_rate
    _eur_rate = None
